torch/onnx/_internal/exporter.py

```diff
@@ -851,18 +860,20 @@ def adapt_torch_inputs_to_onnx(
             >>> onnx_program = torch.onnx.dynamo_export(func_with_nested_input_structure, x_dict, y_tuple)
             >>> print(x_dict, y_tuple)
             {'a': tensor(1.)} (tensor(2.), (tensor(3.), tensor(4.)))
-            >>> print(onnx_program.adapt_torch_inputs_to_onnx(x_dict, y_tuple))
+            >>> print(onnx_program.adapt_torch_inputs_to_onnx(x_dict, y_tuple, model=func_with_nested_input_structure))
             (tensor(1.), tensor(2.), tensor(3.), tensor(4.))
 
         .. warning::
             This API is experimental and is *NOT* backward-compatible.
 
         """
-        return self._input_adapter.apply(*model_args, **model_kwargs)
+        return self._input_adapter.apply(*model_args, model=model, **model_kwargs)
 
     @_beartype.beartype
     def adapt_torch_outputs_to_onnx(
-        self, model_outputs: Any
+        self,
+        model: Union[torch.nn.Module, Callable, torch_export.ExportedProgram],
```

Comment:
For my education, in `ExportedProgram`, when it lifts the buffers and constants, do they maintain a `state_dict` to store them as implicit inputs for users? If they do, it should be accessible from `ExportedProgram`, otherwise, with the same args and kwargs, we can't run it. So, I am wondering why we are moving args "model" from traced to user provided during adapt inputs/outpus. Shouldn't the original traced model have everything?

Reply:
ExportedProgram does have `state_dict`, but they are fakefied when fake tensor is enabled during export. For this scenario (only), we need an external (non-fake) state_dict.

The next PR will make `model` optional to be used only for fake tensor scenario

Reply:
Ah right. Only if we could enable fake mode inside `torch.onnx.dynamo_export`, we can avoid have users provide models twice during export... Do you think it's doable?

Reply:
This way, they model would have to fit in memory to be passed on to `dynamo_export`. Having it externally might open us a way to convert the state_dict directly to onnx external data. This is what I am going to work on after this series of PRs for hte "basic" fake tensor support

Reply:
Make sense. I forgot that.

Reply:
Sorry, just one another Q. So if the model is not loadable or openable, how do we expect users to pass in during adapt inputs/outputs. Where is the crucial stage we facing memory issue?

Reply:
Fake support has 2 stages. 

1. export the model, which is the goal of this PR. For testing, we are assuming the model fits in memory (in a temporary basis)
2. Convert/load a pytorch state_dict/checkpoint into the ONNX model. This may change how input/output adapt works

You questions seems to be the second stage, which we are not tackling just yet, That will be the next task, which might involve serializing a regular pytorch checkpoint directly into ONNX and then merge it with the ONNX file. This way, `adapt_torch_*` API wouldn't need the `model_with_state_dict`